ckanext/xloader/plugin.py

```diff
@@ -183,3 +200,40 @@ def get_helpers(self):
             "xloader_status_description": xloader_helpers.xloader_status_description,
             "is_resource_supported_by_xloader": xloader_helpers.is_resource_supported_by_xloader,
         }
+
+
+def _should_remove_unsupported_resource_from_datastore(res_dict_or_obj):
+    if not toolkit.asbool(toolkit.config.get('ckanext.xloader.clean_datastore_tables', False)):
+        return False
+    if isinstance(res_dict_or_obj, Resource):
+        res_dict_or_obj = res_dict_or_obj.__dict__
+    return ((not XLoaderFormats.is_it_an_xloader_format(res_dict_or_obj.get('format', u''))
+                or res_dict_or_obj.get('url_changed', False))
```

Comment:
I don't see how `url_changed` is relevant in this context.

Reply:
@ThrawnCA 

I had to really really think about this one haha. I did a lot of testing in our own setup initially regarding this. The check for `url_changed` is for the scenario in which a user changes their resource from an upload to a URL but do not update the format

BUT, from reading the `_submit_to_xloader` code, this might just be a thing with our setup in which our system cannot fetch from external sources? So just another case in which I brought over bogus code from my initial code :)

Have removed the `url_changed` check now